File: stvs/src/test/java/edu/kit/iti/formal/stvs/logic/io/xml/roundTripTests/XmlConfigRoundtripTest.java
package edu.kit.iti.formal.stvs.logic.io.xml.roundTripTests;

import edu.kit.iti.formal.stvs.logic.io.ExportException;
import edu.kit.iti.formal.stvs.logic.io.ExporterFacade;
import edu.kit.iti.formal.stvs.logic.io.ImportException;
import edu.kit.iti.formal.stvs.logic.io.ImporterFacade;
import edu.kit.iti.formal.stvs.logic.io.xml.TestUtils;
import edu.kit.iti.formal.stvs.model.config.GlobalConfig;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.io.File;
import java.io.IOException;
import java.net.URISyntaxException;
import java.util.Collection;

import static org.junit.Assert.assertEquals;

/**
 * Created by bal on 03.03.17.
 */
@RunWith(Parameterized.class)
public class XmlConfigRoundtripTest {
  private File file;

  @Parameterized.Parameters
  public static Collection<File> testFiles() throws URISyntaxException {
    return TestUtils.getTestFiles(TestUtils.FileType.CONFIG, TestUtils.Status.VALID);
  }

  public XmlConfigRoundtripTest(File input) {
    this.file = input;
  }

  @Test
  public void configRoundtripTest() throws URISyntaxException, ImportException,
      IOException, ExportException {
    String fileContentsBefore = TestUtils.readFromFile(file.getAbsolutePath());
    GlobalConfig importedConfig = ImporterFacade.importConfig(file, ImporterFacade.ImportFormat
        .XML);
    File tempFile = File.createTempFile("test", "");
    ExporterFacade.exportConfig(importedConfig, ExporterFacade.ExportFormat.XML, tempFile);
    String fileContentsAfter = TestUtils.readFromFile(tempFile.getAbsolutePath());
    assertEquals(TestUtils.removeWhitespace(fileContentsBefore),
        TestUtils.removeWhitespace(fileContentsAfter));
  }
}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/logic/io/xml/roundTripTests/XmlConfigRoundtripTest.kt
/* *****************************************************************
 * This file belongs to verifaps-lib (https://verifaps.github.io).
 * SPDX-License-Header: GPL-3.0-or-later
 * 
 * This program isType free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program isType distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * *****************************************************************/
package edu.kit.iti.formal.stvs.logic.io.xml.roundTripTests

import com.google.common.truth.Truth
import edu.kit.iti.formal.stvs.logic.io.ExporterFacade
import edu.kit.iti.formal.stvs.logic.io.ImporterFacade
import edu.kit.iti.formal.stvs.logic.io.xml.TestUtils
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.io.File

/**
 * Created by bal on 03.03.17.
 */
class XmlConfigRoundtripTest {
    @ParameterizedTest
    @MethodSource("configFiles")
    fun configRoundtripTest(file: File) {
        val fileContentsBefore = file.reader().readText()
        val importedConfig = ImporterFacade.importConfig(file)
        val fileContentsAfter = TestUtils.stringOutputStream {
            ExporterFacade.exportConfig(importedConfig, it)
        }

        Truth.assertThat(TestUtils.removeWhitespace(fileContentsAfter))
            .isEqualTo(TestUtils.removeWhitespace(fileContentsBefore))
    }

    companion object {
        @JvmStatic
        fun configFiles(): Collection<Arguments> =
            TestUtils.getTestFiles(TestUtils.FileType.CONFIG, TestUtils.Status.VALID).map { Arguments.of(it) }
    }
}